### PR TITLE
update doc

### PR DESCRIPTION
**Pull request by jxx-gg**
* added "tunnel" prefix to various places
* added a link in the index.md to the gNMI usage doc
* removed "ack" from the diagram and the doc
* add a description regarding local and tunnel connections at the same time
* removed the description for "capability".

### CODE REVIEW

**Review by gcsl — CHANGES_REQUESTED** [review]

File: doc/grpctunnel_design.md [hunk 1]
```diff
@@ -205,7 +207,7 @@ tunnel streams are used to forward the correct TCP client connections.
 
 #### Data Field
 
-The `data` field is used to encapsulate raw bytes received from the client code.
+The `data` field is used to encapsulate raw bytes received from the tunnel client code.
```
[the remark this answers is not in the document]
Comment (jxx-gg) [reply doc/grpctunnel_design.md]: Done

File: doc/grpctunnel_design.md [hunk 2]
```diff
@@ -296,49 +298,49 @@ handler function will accept. A handler function and register handler are
...
+will accept a connection from a tunnel client. The new tunnel server will then wait for new 
+session requests.
 
-When `NewSession` is called on the server, it will send a register request, with
-an ack to the tunnel client. The client will then check if it can handle the
-request, and if it can, it will create a new tunnel session to the server. The
-tunnel session is then forwarded to the client handler function on the client,
-and returned to the `NewSession` request on the server.
+When `NewSession` is called on the tunnel server, it will send a register request, with
+an ack to the tunnel client. The tunnel client will then check if it can handle the
```
[the remark this answers is not in the document]
Comment (jxx-gg) [reply doc/grpctunnel_design.md]: Done

File: doc/grpctunnel_design.md [hunk 3]
```diff
@@ -296,49 +298,49 @@ handler function will accept. A handler function and register handler are
...
-When `NewSession` is called on the client, it will send a register request to
-the server. The server will check if it can handle the register request via the
-server register handler. This handler will return an error if it cannot handle
-the requested session. If it can handle it, the server will send a register
-session ack back to the client and the client will start a new tunnel stream,
+When `NewSession` is called on the tunnel client, it will send a register request to
+the tunnel server. The tunnel server will check if it can handle the register request via the
+tunnel server register handler. This handler will return an error if it cannot handle
+the requested session. If it can handle it, the tunnel server will send a register
+session ack back to the tunnel client and the tunnel client will start a new tunnel stream,
```
[the remark this answers is not in the document]
Comment (jxx-gg) [reply doc/grpctunnel_design.md]: Done

**Comment by jxx-gg** [comment]
just merged to recent update

**Review by gcsl — APPROVED** [review]

**Pull request merged**